language/gpt3/megatron/dataset.py

```diff
@@ -0,0 +1,162 @@
+import os
+import sys
+
+sys.path.append(os.environ["MEGATRON_PATH"])
+from megatron.tokenizer import build_tokenizer
+from megatron.utils import get_ltor_masks_and_position_ids
+import torch
+import argparse
+import utils
+import json
+
+from argparse import Namespace
+
+
+PROMPT_DICT = {
+    "prompt_input": (
+        "{instruction}{input}"
+    )
+}
+
+
+class Dataset:
+    def __init__(
+        self,
+        dataset_path,
+        batch_size=1,
+        args=Namespace(),
+        gen_kwards={},
+        total_count_override=None,
+        perf_count_override=None,
+        debug=False,
+    ):
+        print("Constructing QSL")
+
+        self.dataset = "cnn_dailymail"
+        self.dataset_path = dataset_path
+        self.batch_size = batch_size
+        self.debug = debug
+        self.gen_kwards = gen_kwards
+
+        ## TODO: provide arguments in command line
+        args.rank = 0
+        args.tokenizer_type = "SentencePieceTokenizer"
+        args.make_vocab_size_divisible_by = 128
+        args.tensor_model_parallel_size = 8
+        args.vocab_extra_ids = 0
+        args.tokenizer_model = "./data/c4_en_301_5Mexp2_spm.model"
+        self.tokenizer = build_tokenizer(args)
+
+        self.list_data_dict = utils.jload(self.dataset_path)
+        self.max_input_tokens = 2048 - self.gen_kwards.get("max_new_tokens", 128)
```

Comment:
I don't think this is correct. The max_input_tokens as suggested by Google is 2048, combined with max new tokens of 128